rtsf-at-checkout-device-scale/driver/scaleDriver.go

```diff
@@ -165,14 +141,33 @@ func findSerialPort(pid string, vid string) (string, error) {
 			}
 		}
 	}
-	return "", fmt.Errorf("Serial device with pid:vid %s:%s not found", pid, vid)
+	return "", fmt.Errorf("serial device with pid:vid %s:%s not found", pid, vid)
 }
 
 // AddDevice is a callback function that is invoked
 // when a new Device associated with this Device Service is added
 func (drv *ScaleDriver) AddDevice(deviceName string, protocols map[string]models.ProtocolProperties, adminState models.AdminState) error {
-	// Nothing required to do for AddDevice since new devices will be available
-	// when data is posted to REST endpoint
+	// get device definition from sdk, then from device definition, get PID and VID from the protocol section
+	serialProtocol := protocols["serial"]
+	serialPort, err := findSerialPort(serialProtocol["PID"], serialProtocol["VID"])
+
+	driver.lc.Debugf("[serialPort]: %v, err: %v", serialPort, err)
```

Comment:
I don't like this since one or the other will not be set. I think it can be removed since both cases are handled below.